src/main/src/main/java/org/geoserver/config/GeoServerResourcePersister.java

```diff
@@ -143,40 +108,6 @@ public void handleRemoveEvent(CatalogRemoveEvent event) {
         }
     }
 
-    /**
-     * Determine unique name of the form <code>newName.extension</code>. newName will have a number
-     * appended as required to produce a unique resource name.
-     *
-     * @param resource Resource being renamed
-     * @param newName proposed name to use as a template
-     * @param extension extension
-     * @return New UNDEFINED resource suitable for use with rename
-     * @throws IOException If unique resource cannot be produced
-     */
-    private Resource uniqueResource(Resource resource, String newName, String extension)
```

Comment:
I don't quite agree with this removal - even if it isn't currently used anywhere other then style, it may be better to keep it in ResourcePersister and make it public rather than move the logic to CatalogImpl - if there is the need to rename some other type of resource in the future, this has a better chance of keeping all the logic in one place.

Reply:
ResourcePersister has a specific function of handling catalog events, it is not a class to expose  utility methods to other classes. I would suggest moving it to the "Resources" class then.

Reply:
I have added a commit to do so

Reply:
Great, that works for me